rfcs/20230807-unify-primitive-and-graph-verbose/README.md

```diff
@@ -0,0 +1,173 @@
+onednn_graph_verbose,exec,cpu,100006,convolution_post_ops,CONV_0;BINARY_1,data:NCX; filter:OIX;;,in0_f32:0:strided:undef:50x64x56x56:200704s3136s56s1 in1_f32:1:strided:constant:64x64x1x1:64s1s1s1 in2_f32:2:strided:constant:64:1 in3_f32:3:strided:undef:1x1x1x1:1s1s1s1 out0_f32:2209:strided:undef:50x64x56x56:200704s3136s56s1,fpm:strict,dnnl_backend,18.824
+```
+
+## Motivation
+
+oneDNN Graph API feature has been enabled and built by default in oneDNN's build
+system and it is general available since oneDNN v3.1 release. From the user
+perspective, a single unified verbose interface can be easier and
+straightforward to control the verbose mode for both primitive and graph APIs.
+From the implementation perspective, unifying them will save lots of duplicated
+codes in the graph API implementation and it will be better to maintain and
+align with the changes from primitive verbose.
+
+## Proposal
+
+This proposal is to unify the user interface, verbose format and verbose header
+with primitive verbose.
+
+### Unified User Interface
+
+To better unify the user interface with primitive verbose, this proposal is to
+leverage `ONEDNN_VERBOSE`:
+
+- `ONEDNN_GRAPH_VERBOSE`  will be deprecated and graph verbose can be controlled
+  via `ONEDNN_VERBOSE`
+- When `ONEDNN_VERBOSE=1`, verbose of both primitive and compiled partition exec
+  profiling will be printed
+- When `ONEDNN_VERBOSE=2`, verbose of both primitive creation/execution,
+  partition compilation and compiled partition execution profiling will be
+  printed
+- string based flag `profile`, `profile_exec`, `profile_create`, `check`,
+  `error` and `debuginfo` can be used to control verbose for both primitive and
+  graph API
+
+Verbose from primitive and graph API will also be printed if user set the
+verbose level through `dnnl_set_verbose` API.
+
+To achieve this, `get_verbose()` and `verbose_t` need to be extended and
+`get_graph_verbose()` will be removed.
+
+### Verbose Filter
+
+Sometimes, users may only need the graph or primitive verbose. To achieve
+this, this proposal introduces verbose filter. User can specify the filter along
+with the verbose flag. e.g. `ONEDNN_VERBOSE=profile,check,filter=xxx`.
+
+Since oneDNN verbose is designed to be used by frontend users, this proposal
+prefers to support filtering the verbose by the components from public API, such
+as `primitive`, `graph`, `convolution`, `matmul`, etc. Currently, the
+implementation of this RFC have an initial support for filtering verbose from
+`primitive` and `graph` API. e.g.
+
+- `ONEDNN_VERBOSE=profile_exec,filter=graph` will print verbose of
+  compiled_partition execution profiling from graph API
+- `ONEDNN_VERBOSE=check,filter=graph+primitive` will print verbose of check info
```

Comment:
I would recommend the argument of filter to be a regular expression.
Internally, we would then parse that regular expression and match it to components upon reading the environment variable.

Reply:
thanks for the suggestion. fixed:)